moto/applicationautoscaling/models.py

```diff
@@ -0,0 +1,223 @@
+        self.region = region
+        self.ecs_backend = ecs
+        self.targets = OrderedDict()
+
+    def reset(self):
+        region = self.region
+        ecs = self.ecs_backend
+        self.__dict__ = {}
+        self.__init__(region, ecs)
+
+    @property
+    def applicationautoscaling_backend(self):
+        return applicationautoscaling_backends[self.region]
+
+    def delete_scaling_policy(self):
+        """ Not yet implemented. """
+        pass
+
+    def delete_scheduled_action(self):
+        """ Not yet implemented. """
+        pass
+
+    def deregister_scalable_target(self):
+        """ Not yet implemented. """
+        pass
+
+    def describe_scalable_targets(
+        self, namespace, r_ids=None, dimension=None,
+    ):
+        """ Describe scalable targets. """
+        if r_ids is None:
+            r_ids = []
+        targets = self._flatten_scalable_targets(namespace)
+        if dimension is not None:
+            targets = [t for t in targets if t.scalable_dimension == dimension]
+        if len(r_ids) > 0:
+            targets = [t for t in targets if t.resource_id in r_ids]
+        return targets
+
+    def _flatten_scalable_targets(self, namespace):
+        """ Flatten scalable targets for a given service namespace down to a list. """
+        targets = []
+        for dimension in self.targets.keys():
+            for resource_id in self.targets[dimension].keys():
+                targets.append(self.targets[dimension][resource_id])
+        targets = [t for t in targets if t.service_namespace == namespace]
+        return targets
+
+    def describe_scaling_activities(self):
+        """ Not yet implemented. """
+        pass
+
+    def describe_scaling_policies(self):
+        """ Not yet implemented. """
+        pass
+
+    def describe_scheduled_actions(self):
+        """ Not yet implemented. """
+        pass
+
+    def generate_presigned_url(self):
+        """ Not yet implemented. """
+        pass
+
+    def get_waiter(self):
+        """ Not yet implemented. """
+        pass
+
+    def put_scaling_policy(self):
+        """ Not yet implemented. """
+        pass
+
+    def put_scheduled_action(self):
+        """ Not yet implemented. """
+        pass
+
+    def register_scalable_target(self, namespace, r_id, dimension, **kwargs):
+        """ Registers or updates a scalable target. """
+        _ = _target_params_are_valid(namespace, r_id, dimension)
+        if namespace == ServiceNamespaceValueSet.ECS.value:
+            _ = self._ecs_service_exists_for_target(r_id)
+        if self._scalable_target_exists(r_id, dimension):
+            target = self.targets[dimension][r_id]
+            target.update(kwargs)
+        else:
+            target = FakeScalableTarget(self, namespace, r_id, dimension, **kwargs)
+            self._add_scalable_target(target)
+        return target
+
+    def _scalable_target_exists(self, r_id, dimension):
+        exists = False
```

Comment:
This can be simplified to `return r_id in self.targets.get(dimension, [])` if I'm not mistaken?

Reply:
Yep, definitely, will fix. I was originally storing the scalable targets in a more complex data structure that involved more code in this function, I then simplified the data structure but didn't simplify this function accordingly.

Reply:
Fixed.